src/Service/Models/SigninTokenRequest.cs

```diff
@@ -1,6 +1,10 @@
 namespace Passwordless.Service.Models;
 
-public class SigninTokenRequest(string userId) : RequestBase
+public class SigninTokenRequest(string userId, int? timeToLive = null) : RequestBase
 {
     public string UserId { get; } = userId;
+    /// <summary>
+    /// Time to live is the number of seconds the token has before it expires.
+    /// </summary>
+    public int? TimeToLive { get; } = timeToLive;
```

Comment:
I disagree. This is the request model. In order for it to be a timespan, the user would have to post "00:00:13" as a string for a 13 second timespan. IMO, it would be much easier to say 13 seconds as a time to live. But I could be convinced otherwise.

Reply:
We can make a converter that supports both `13` and `00:00:13` as input. The issue is that sooner or later `obj.TimeToLive` will surface somewhere in code, and it will not be clear what units it operates with.

Reply:
As an intermediate step, i'll create a DTO that takes the request object and converts it to a timespan. Later we can add that converter if we feel its needed.

Reply:
I agree that the API is nicer with seconds. I think the intermediate step of converting to span seems reasonable.

Reply:
In that case, how about this:

```csharp
[JsonPropertyName("timeToLive")]
[Obsolete("This property is only used for serialization")]
public int? TimeToLiveSeconds { get; }

[JsonIgnore]
public TimeSpan TimeToLive { get; } = TimeSpan.FromSeconds(TimeToLiveSeconds ?? 120);
```

Reply:
But why is that timespan assuming a default value? Seems odd.

Reply:
Should this be on a new type farther down?

Something like this:
```
using Passwordless.Service.Models;

namespace Passwordless.Service.MagicLinks.Models;

public class MagicLinkToken : VerifySignInToken
{
    private static readonly TimeSpan DefaultTimeToLive = TimeSpan.FromHours(1);

    private static TimeSpan GetTimeToLive(MagicLinkTokenRequest request) => request.TimeToLive ?? DefaultTimeToLive;
    
    public static MagicLinkToken GetInstance(MagicLinkTokenRequest request, DateTimeOffset now) => new()
    {
        Success = true,
        UserId = request.UserId,
        Timestamp = now.UtcDateTime,
        ExpiresAt = now.UtcDateTime.Add(GetTimeToLive(request)),
        TokenId = Guid.NewGuid(),
        Type = "magic_link",
        RpId = request.RPID,
        Origin = request.Origin
    };
}
```

That way the default value is encapsulated by the actual token it is for, instead of being a part of the request.

Reply:
Seems like a bit too much indirection for my "keep it simple stupid" taste

Reply:
I think Oleksii's approach is fine